pkgs/by-name/st/stash/package.nix

```diff
@@ -1,50 +1,138 @@
+    rev = "refs/tags/v${version}";
+    hash = srcHash;
   };
-in
-stdenv.mkDerivation (finalAttrs: {
-  inherit version;
 
-  pname = "stash";
+  ui = stdenv.mkDerivation (final: {
+    inherit version;
+    pname = "${pname}-ui";
+    src = "${src}/ui/v2.5";
+
+    yarnOfflineCache = fetchYarnDeps {
+      yarnLock = "${final.src}/yarn.lock";
+      hash = yarnHash;
+    };
 
-  src = fetchurl { inherit (sources.${stdenv.system}) url hash; };
+    nativeBuildInputs = [
+      yarnConfigHook
+      yarnBuildHook
+      # Needed for executing package.json scripts
+      nodejs
+    ];
 
-  dontUnpack = true;
+    postPatch = ''
+      substituteInPlace codegen.ts \
+        --replace-fail "../../graphql/" "${src}/graphql/"
+    '';
 
-  installPhase = ''
-    runHook preInstall
+    buildPhase = ''
+      runHook preBuild
 
-    install -Dm755 $src $out/bin/stash
+      export HOME=$(mktemp -d)
+      export VITE_APP_DATE='1970-01-01 00:00:00'
+      export VITE_APP_GITHASH=${gitHash}
+      export VITE_APP_STASH_VERSION=v${version}
+      export VITE_APP_NOLEGACY=true
 
-    runHook postInstall
+      yarn --offline run gqlgen
+      yarn --offline build
+
+      mv build $out
+
+      runHook postBuild
+    '';
+
+    dontInstall = true;
+    dontFixup = true;
+  });
+in
+(buildGoModule {
+  inherit
+    pname
+    src
+    version
+    vendorHash
+    ;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X 'github.com/stashapp/stash/internal/build.buildstamp=1970-01-01 00:00:00'"
+    "-X 'github.com/stashapp/stash/internal/build.githash=${gitHash}'"
+    "-X 'github.com/stashapp/stash/internal/build.version=v${version}'"
+    "-X 'github.com/stashapp/stash/internal/build.officialBuild=false'"
+  ];
+  tags = [
+    "sqlite_stat4"
+    "sqlite_math_functions"
+  ];
+
+  subPackages = [ "cmd/stash" ];
+
+  preBuild = ''
+    cp -a ${ui} ui/v2.5/build
+    # `go mod tidy` requires internet access and does nothing
+    echo "skip_mod_tidy: true" >> gqlgen.yml
+    # remove `-trimpath` fron `GOFLAGS` because `gqlgen` does not work with it
+    GOFLAGS="''${GOFLAGS/-trimpath/}" go generate ./cmd/stash
   '';
 
-  meta = with lib; {
-    description = "Stash is a self-hosted porn app";
-    homepage = "https://github.com/stashapp/stash";
-    license = licenses.agpl3Only;
-    maintainers = with maintainers; [ Golo300 ];
-    platforms = builtins.attrNames sources;
+  strictDeps = true;
+
+  meta = {
     mainProgram = "stash";
+    description = "Organizer for your adult videos/images";
+    license = lib.licenses.agpl3Only;
+    homepage = "https://stashapp.cc/";
+    changelog = "https://github.com/stashapp/stash/blob/v${version}/ui/v2.5/src/docs/en/Changelog/v${lib.versions.major version}${lib.versions.minor version}0.md";
+    maintainers = with lib.maintainers; [
+      Golo300
+      DrakeTDL
+    ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
   };
-})
+}).overrideAttrs
```

Comment:
any particular reason for this overrideAttrs? can put `passthru` in buildGoModule itself.

also maybe add `passthru.frontend = ui;` so people have the ability to override ui build from outside.

Reply:
> any particular reason for this overrideAttrs? can put `passthru` in buildGoModule itself.

I initially did that because buildGoModule didn't accept a fixed-point function and I didn't think that I could request itself from the arguments. Don't know why I thought that since it is a package like any other.

As suggested, I inlined `passthru` and inherited frontend.